frontend/src/metabase/dashboard/containers/DashboardSharingEmbeddingModal/DashboardSharingEmbeddingModal.tsx

```diff
@@ -38,15 +40,18 @@ export const DashboardSharingEmbeddingModal = (
   const updateDashboardEmbeddingParams = (embeddingParams: EmbedOptions) =>
     dispatch(updateEmbeddingParams(dashboard, embeddingParams));
 
-  const getPublicUrl = ({ public_uuid }: { public_uuid: string }) =>
-    Urls.publicDashboard(public_uuid);
+  const getPublicUrl = ({ public_uuid }: EmbedResource) => {
```

Comment:
I gave a similar comment in `frontend/src/metabase/query_builder/components/QuestionEmbedWidget/QuestionEmbedWidget.tsx` Please take a look there.

tl;dr I think we should keep the signature as is `public_uuid` to always be string.

Reply:
I think ideally we should do that, but `Dashboard` and `Card` both have `public_uuid` typed as `string | null` so I'm not sure how much we can do before typescript complains. I think we would need to check in a parent component if we have to do a null check